CI: Add GPU Vs CPU dry HS comparison job

valeriabarra opened the pull request:
## Purpose 
The purpose of this PR is to add a CI job for the GPU Vs CPU comparison for the dry HS test case.

Closes #1926 


## Content
- Renamed the existing GPU and comparison job for dry BW
- Added a GPU and comparison job for the dry HS test case

Review checklist

I have:
- followed the codebase contribution guide: https://clima.github.io/ClimateMachine.jl/latest/Contributing/
- followed the style guide: https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/
- followed the documentation policy: https://github.com/CliMA/policies/wiki/Documentation-Policy
- checked that this PR does not duplicate an open PR.

In the Content, I have included 
- relevant unit tests, and integration tests, 
- appropriate docstrings on all functions, structs, and modules, and included relevant documentation.


----
- [x] I have read and checked the items on the review checklist.

valeriabarra commented:
bors try

valeriabarra commented:
The plots look identical: 

- GPU dry HS high-top: 
    - `rho`, day 8, level 10:
![rho_day8_level10](https://github.com/CliMA/ClimaAtmos.jl/assets/39932030/3d1b219c-d168-4429-b9c9-edf4f93af321)


   - `rho_e_tot`, day 8, level 10:
![rho_e_tot_day8_level10](https://github.com/CliMA/ClimaAtmos.jl/assets/39932030/337428dc-606f-49e3-aed3-e8bff1063dc2)


   - `u`, day 8, level 10:
![u_day8_level10](https://github.com/CliMA/ClimaAtmos.jl/assets/39932030/a5408808-a148-496b-81bd-3462455c52eb)


   - `w`, day 8, level 10+half:
![w_day8_level10](https://github.com/CliMA/ClimaAtmos.jl/assets/39932030/7b4cf9ce-2e7b-4a4c-b81e-de4fc5dcc1a2)


- CPU dry HS high-top: 
    - `rho`, day 8, level 10:
![rho_day8_level10](https://github.com/CliMA/ClimaAtmos.jl/assets/39932030/d8389372-ae48-4510-af74-f250ef863a2f)


   - `rho_e_tot`, day 8, level 10:
![rho_e_tot_day8_level10](https://github.com/CliMA/ClimaAtmos.jl/assets/39932030/08d0fc40-62cc-4305-b534-1568cd395737)



   - `u`, day 8, level 10:
![u_day8_level10](https://github.com/CliMA/ClimaAtmos.jl/assets/39932030/b4d1dbe3-5666-476c-af1e-0a66d6a36b02)



   - `w`, day 8, level 10+half:
![w_day8_level10](https://github.com/CliMA/ClimaAtmos.jl/assets/39932030/099c58c9-80d3-48ec-9050-700f4268c123)

valeriabarra commented:
@charleskawczynski I'm not sure why CI failed. I only slightly renamed the existing dry BW job and added the new HS comparison job. 

I can run the comparison just fine locally on my machine and it looks similar to the dry BW test case you added.

charleskawczynski commented:
> @charleskawczynski I'm not sure why CI failed. I only slightly renamed the existing dry BW job and added the new HS comparison job.

It timed out, waiting for gpu resources.

charleskawczynski commented on .buildkite/pipeline.yml:
```diff
@@ -805,20 +805,35 @@ steps:
     steps:
 
-      - label: "GPU: GPU implicit baro wave"
+      - label: "GPU: GPU implicit dry baro wave"
```
Can we just leave out the "dry" for the dry case, and add moist for the moist case? I think the naming scheme is easier to follow if we leave out default values (dry is the default)

valeriabarra replied:
Sure we can! I will change it

valeriabarra commented:
bors r+

valeriabarra commented:
@charleskawczynski the time-out on the GPU comparison jobs happened again. This doesn't seem accidental now. Did I do something wrong in the renaming? Do you spot anything?

valeriabarra commented:
bors r+

charleskawczynski commented:
> This doesn't seem accidental now. Did I do something wrong in the renaming? Do you spot anything?

It might be easier to review if this PR only added new jobs, and didn't also modify existing ones.

valeriabarra commented:
> > This doesn't seem accidental now. Did I do something wrong in the renaming? Do you spot anything?
> 
> It might be easier to review if this PR only added new jobs, and didn't also modify existing ones.

Okay, it was just a simple renaming (and I had made sure that I changed job id's and artifact paths accordingly). But I reverted it for simplicity.

charleskawczynski commented:
No problem, this looks fine, except we need to add the `sphere_held_suarez_rhoe_hightop` key for that job (it currently only has a label). This is why it previously hung

valeriabarra commented:
bors r+

valeriabarra commented:
bors r+

valeriabarra commented:
bors r+

valeriabarra commented:
bors r+
